parsers/AU-LH.py

```diff
@@ -0,0 +1,200 @@
+        "consumption": cleaned_values[0],
+        "solar": cleaned_values[1],
+        "battery": cleaned_values[2],
+        "diesel": cleaned_values[3],
+    }
+
+    # determine the sign of the battery value based on the other values
+    if value_dict["diesel"] + value_dict["solar"] < value_dict["consumption"]:
+        value_dict["battery"] *= -1
+
+    return value_dict
+
+
+def fetch_production(
+    zone_key: ZoneKey,
+    session: HTMLSession = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> Union[List[dict], dict]:
+    if target_datetime is not None:
+        raise ParserException(
+            PARSER_NAME, "This parser is not yet able to parse past dates", zone_key
+        )
+
+    is_new_session = session is None
+    if is_new_session:
+        session = HTMLSession()
+
+    data = fetch_data(zone_key, session)
+
+    production_list = ProductionBreakdownList(logger=logger)
+    production_list.append(
+        zoneKey=zone_key,
+        datetime=datetime.now(timezone.utc),
+        production=ProductionMix(
+            gas=data["diesel"],
+            solar=data["solar"],
+        ),
+        storage=StorageMix(battery=data["battery"]),
+        source=DATA_URL,
+    )
+
+    if is_new_session:
+        session.close()
+
+    return production_list.to_list()
+
+
+def fetch_consumption(
+    zone_key: ZoneKey,
+    session: HTMLSession = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> Union[Dict[str, Any], List[Dict[str, Any]]]:
+    if target_datetime is not None:
+        raise ParserException(
+            PARSER_NAME, "This parser is not yet able to parse past dates", zone_key
+        )
+
+    is_new_session = session is None
+    if is_new_session:
+        session = HTMLSession()
+
+    data = fetch_data(zone_key, session)
+
+    consumption_list = TotalConsumptionList(logger=logger)
+    consumption_list.append(
+        zoneKey=zone_key,
+        datetime=datetime.now(timezone.utc),
+        consumption=data["consumption"],
+        source=DATA_URL,
+    )
+
+    if is_new_session:
+        session.close()
+
+    return consumption_list.to_list()
+
+
+def fetch_total_production(
+    zone_key: ZoneKey,
+    session: HTMLSession = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> Union[dict, List[dict]]:
+    if target_datetime is not None:
+        raise ParserException(
+            PARSER_NAME, "This parser is not yet able to parse past dates", zone_key
+        )
+
+    is_new_session = session is None
+    if is_new_session:
+        session = HTMLSession()
+
+    data = fetch_data(zone_key, session)
+
+    production_list = TotalProductionList(logger=logger)
+
+    production_list.append(
+        zoneKey=zone_key,
+        datetime=datetime.now(timezone.utc),
+        value=data["solar"] + data["diesel"],
+        source=DATA_URL,
+    )
+
```

Comment:
This is not really what this data type is supposed to return. It is ususally used when the total generation/production is reported seperatly from the individual production modes.

It is also missing battery discharge.

Reply:
Ok, I'll remove it